Sources/SourceControl/GitRepository.swift

```diff
@@ -27,8 +27,11 @@ private struct GitShellHelper {
     /// Private function to invoke the Git tool with its default environment and given set of arguments.  The specified
     /// failure message is used only in case of error.  This function waits for the invocation to finish and returns the
     /// output as a string.
-    func run(_ args: [String], environment: [String: String] = Git.environment) throws -> String {
-        let process = Process(arguments: [Git.tool] + args, environment: environment, outputRedirection: .collect)
+    func run(_ args: [String],
+             environment: [String: String] = Git.environment,
+             outputRedirection: Process.OutputRedirection = .collect) throws -> String
+    {
+        let process = Process(arguments: [Git.tool] + args, environment: environment, outputRedirection: outputRedirection)
```

Comment:
`ProcessResult` does not contain the output if the output is redirected...

Reply:
Some .streamAndCollect option might be one possible solution

Reply:
I see — I'll take a closer look and see if what I can suggest here.  Do you know whether this is something that changed through other commits in the meantime, while this PR was up?

Reply:
Having a `.streamAndCollect` does indeed look interesting — I guess in this case you want to stream `stderr` but collect `stdout`.

I almost seems to me as if stream and collect should be booleans, not exclusive choices — with `.none`, `.stream`, and `.collect` as already being options, adding `.streamAndCollect` seems like a lesser choice than just letting each of the outputs have both an optional streaming closure as well as an optional collection container, independently controllable.

Reply:
Actually, in this case `GitRepository.callGit()` could collect the data in the stream closure, and then add it to the `GitCloneError` instead of having a whole `ProcessResult` as the third parameter.